### PR TITLE
Eventing docs updates for 6.5

### PR DESCRIPTION
**Pull request by amarantha-k**

### CODE REVIEW

File: modules/eventing/pages/eventing-faq.adoc [hunk 1]
```diff
@@ -199,28 +199,38 @@ Function via a binding.  It is also possible to utilize the cURL API to read add
...
 
-* When I schedule a timer to fire at an exact time, I see some delay why?
+* Timer Delays: When I schedule a timer to fire at an exact time, I see some delay why?
 +
 The timer implementation is designed to handle large numbers of distributed timers (i.e., millions of timers) and the only promise is to run timers as soon as possible, e.g. no timers lost.
 +
 In a steady state you may see a 3-4 second delay from the scheduled time, however if scheduling timers close to the system wall-clock this delay may increase to about 14 seconds.  
 For more details on Timer scheduling refer to xref:eventing-timers.adoc#wall-clock-accuracy[Timers: Wall-clock Accuracy] section.
 
+* Can I Cancel a Timer
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-faq.adoc]:
```suggestion
* Can I cancel a timer? 
```

Question and title case for consistency

File: modules/eventing/pages/eventing-faq.adoc [hunk 2]
```diff
@@ -199,28 +199,38 @@ Function via a binding.  It is also possible to utilize the cURL API to read add
...
 The timer implementation is designed to handle large numbers of distributed timers (i.e., millions of timers) and the only promise is to run timers as soon as possible, e.g. no timers lost.
 +
 In a steady state you may see a 3-4 second delay from the scheduled time, however if scheduling timers close to the system wall-clock this delay may increase to about 14 seconds.  
 For more details on Timer scheduling refer to xref:eventing-timers.adoc#wall-clock-accuracy[Timers: Wall-clock Accuracy] section.
 
+* Can I Cancel a Timer 
++
+No this is not supported in 6.5, however this will be supported in 6.6.0 - refer to xref:eventing-timers.adoc#limitations[Timers: Limitations].
+
+* Can I create a Recurring timer
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-faq.adoc]:
```suggestion
* Can I create a Recurring timer?
```

Question for consistency

File: modules/eventing/pages/eventing-faq.adoc [hunk 3]
```diff
@@ -199,28 +199,38 @@ Function via a binding.  It is also possible to utilize the cURL API to read add
...
 
+* Can I Cancel a Timer 
++
+No this is not supported in 6.5, however this will be supported in 6.6.0 - refer to xref:eventing-timers.adoc#limitations[Timers: Limitations].
+
+* Can I create a Recurring timer 
++
+No this is not supported in 6.5, however this will be supported in 6.6.0 -  xref:eventing-timers.adoc#limitations[Timers: Limitations].
+
+* Can I schedule Timer far into the future
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-faq.adoc]:
```suggestion
* Can I schedule a Timer far into the future?
```

Question for consistency

File: modules/eventing/pages/eventing-faq.adoc [hunk 4]
```diff
@@ -199,28 +199,38 @@ Function via a binding.  It is also possible to utilize the cURL API to read add
 Yes.
 For example, you can your enrich or update a document with data from another document (using a document id) from any other bucket that is exposed to the Function via a binding with access level of "Read Write" inclusive of the source bucket.
 
-== Timer Delays
+== Timer Behavior
 
-* When I schedule a timer to fire at an exact time, I see some delay why?
+* Timer Delays: When I schedule a timer to fire at an exact time, I see some delay why?
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-faq.adoc]:
```suggestion
* Timer Delays: When I schedule a timer to fire at an exact time, I see some delay. Why?
```

File: modules/eventing/pages/eventing-faq.adoc [hunk 5]
```diff
@@ -122,7 +122,7 @@ When debugging a function a single mutation is blocked and handed off to the deb
 * Can the logic in a Function in production be altered while it is running?
 +
 Yes.
-Functions can be "paused, edited and resumed" without losing a single mutation; as such continuity is maintained. Note this sequence is simailar to the operations of "undeploy, edited, and deployed with a feed boundary ‘From now’", however inthis later case you can lose mutations.
+Functions can be "paused, edited and resumed" without losing a single mutation; as such continuity is maintained. Note this sequence is similar to the operations of "undeploy, edited, and deployed with a feed boundary ‘From now’", however in this later case you can lose mutations.
 
 
 * How to perform Functions lifecycle operations from CI/CD?
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-faq.adoc]:
```suggestion
* How do I perform a Function's lifecycle operations from CI/CD?
```

Grammar check

File: modules/eventing/pages/eventing-faq.adoc [hunk 6]
```diff
@@ -229,9 +239,10 @@ operation fails when a Function lifecycle operation is currently in progress. Li
...
+* How do I increase performance of an Eventing Function?
 +
-You can either scale up vertically by adding additional workers (in the handler's settings) to increase performance  
+You can scale up vertically by adding additional workers (in the handler's settings) to increase performance  
 for a specific Function. 
-You can also scale out horizontally via Couchbase’s elastic scaling option by adding another node and rebalancing.  
-In this case each eventing node is assigned a subset of vBuckets note this increases overall performance for all Functions.
+You can also scale out horizontally via Couchbase’s elastic scaling option by adding another node and rebalancing.  In this case each eventing node is assigned a subset of vBuckets note this sharding increases overall performance for all Functions.
++
+However keep in mind that sometimes the Function is limited by the overall performance of the Data Service in this case it is appropriate to scale the Data service.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-faq.adoc]:
```suggestion
However keep in mind that sometimes the Function is limited by the overall performance of the Data Service. In this case it is appropriate to scale the Data service.
```

File: modules/eventing/pages/eventing-overview.adoc [hunk 7]
```diff
@@ -92,27 +92,36 @@ The following table compares the implementation of the Eventing Service with the
...
+== *Couchbase Eventing Functions*
 
-Couchbase Functions, also referred to as Functions, offer a computing paradigm that developers can use to handle data changes.
-In the Couchbase cluster, you can use the Functions to handle data-changes according to an Event-Condition-Action model.
+Couchbase Eventing Functions, also referred to as Functions or Handlers, offer a computing paradigm that developers can use to handle data changes.
+In the Couchbase cluster, you can use the Functions to process and respond to data-changes according to an Event-Condition-Action model.
 
 For application developers, Functions offer a platform in which you can focus on building business logic, rather than configuration or infrastructure.
-Functions handle and process events that are generated when documents are created, updated, or deleted.
+Functions handle and process events that are generated when documents are created, updated, expired or deleted.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-overview.adoc]:
```suggestion
Functions handle and process events that are generated when documents are created, updated, expired, or deleted.
```

Oxford comma (style rule)

File: modules/eventing/pages/eventing-overview.adoc [hunk 8]
```diff
@@ -92,27 +92,36 @@ The following table compares the implementation of the Eventing Service with the
...
+
 You can configure Functions to generate threshold-based alerts.
 If your business logic requires you to monitor specific parameters inside a document, then you can use Functions to trigger an alert upon logic such as a threshold breach.
 
-Apart from notifications and alerts, Functions provide an option to propagate data changes.
-You can propagate data changes and move documents to a different bucket inside your Couchbase cluster.
-Since Couchbase is a NoSQL document-oriented data platform, Functions provide an easy option to perform cascade deletes.
+Additionally as part of Functions you can
+
+* Utilize N1QL queries ot statements form within a Function
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-overview.adoc]:
```suggestion
* Utilize N1QL queries ot statements form within a Function.
```

Period for consistency with other bullet points.

File: modules/eventing/pages/eventing-overview.adoc [hunk 9]
```diff
@@ -198,13 +207,14 @@ Gaming
...
+The use of MDS services in Couchbase enables workload isolation and independent scalability of any service in a Couchbase cluster. 
+Like Data, Query, GSI, FTS, and Analytics, the Eventing Service supports Multi-Dimensional Scaling (MDS).  
+Adding a new Eventing node is a simple process allowing your business logic to scale in addition the Eventing Service integrates seamlessly 
+with other Couchbase services such as the Data, Query, GSI, FTS, and Analytics nodes. 
 
 .Eventing Service Onboarding Information
-image::eventing-service-onboarding-information.jpg[,570]
+image::eventing-service-onboarding-information.jpg[,100%]
 
 Following are a few aspects during the onboarding process:
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-overview.adoc]:
```suggestion
Following are a few aspects of the onboarding process:
```

File: modules/eventing/pages/eventing-statistics.adoc [hunk 10]
```diff
@@ -8,8 +8,7 @@ Eventing Statistics, for each deployed Function, can be fetched from an Eventing
 
 Eventing Statistics, for each deployed Function, can be fetched from an Eventing node using the Web Console.
 
-To fetch Eventing Statistics for a deployed Function, from the *Couchbase Web Console* > *Servers* page, now click *Statistics.* This page will only show deployed Eventing functions, you can scroll to any section labeled +
-"Eventing Stats: _deployed_function_name_" and expand the section to show the four graphs listed below:
+To fetch Eventing Statistics for a deployed Function you need to leave the Eventing main UI page, first goto the *Couchbase Web Console* > *Servers* page, then click *Statistics.* Note, this page will only show deployed Eventing functions, you can scroll down to any section labeled *Eventing Stats: _deployed_function_name_* and expand the section to show the four key graphs for the Function along with numeric values listed below each graph:
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-statistics.adoc]:
```suggestion
To fetch Eventing Statistics for a deployed Function you need to leave the Eventing main UI page. First go to the *Couchbase Web Console* > *Servers* page, then click *Statistics.* Note that this page will only show deployed Eventing functions. You can scroll down to any section labeled *Eventing Stats: _deployed_function_name_* and expand the section to show the four key graphs for the Function, along with numeric values listed below each graph:
```

Edited slightly for readability

File: modules/eventing/pages/eventing-timers.adoc [hunk 11]
```diff
@@ -32,7 +32,9 @@ In the createTimer syntax:
...
-* reference - This is a unique string that must be passed in to help identify the timer that is being created. References are always scoped to the function and callback they are used with and need to be unique only within this scope. The call returns the reference string if timer was created successfully. If multiple timers are created with the same unique reference, old timers with the same unique reference are implicitly cancelled. If the reference parameter is set to JavaScript null value, a unique reference will be generated.
+* reference - This is a unique string that must be passed in to help identify the timer that is being created. References are always scoped to the function and callback they are used with and need to be unique only within this scope. You may not resue the reference if there is an active timer with the same reference within the current scope. The call returns the reference string if timer was created successfully.  If the reference parameter is set to JavaScript null value, a unique reference will be generated.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-timers.adoc]:
```suggestion
* reference - This is a unique string that must be passed in to help identify the timer that is being created. References are always scoped to the function and callback they are used with and need to be unique only within this scope. You may not reuse the reference if there is an active timer with the same reference within the current scope. The call returns the reference string if timer was created successfully.  If the reference parameter is set to JavaScript null value, a unique reference will be generated. 
```

Spelling (resue → reuse)

File: modules/eventing/pages/eventing-timers.adoc [hunk 12]
```diff
@@ -32,7 +32,9 @@ In the createTimer syntax:
...
+* reference - This is a unique string that must be passed in to help identify the timer that is being created. References are always scoped to the function and callback they are used with and need to be unique only within this scope. You may not resue the reference if there is an active timer with the same reference within the current scope. The call returns the reference string if timer was created successfully.  If the reference parameter is set to JavaScript null value, a unique reference will be generated. 
++
+_Note in release 6.6.0 Eventing Timers can be cancelled using cancelTimer() function, or by creating a new Timer with the same reference as an existing Timer refer to xref:eventing-timers.adoc#limitations[Timers: Limitations]._
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-timers.adoc]:
```suggestion
_Note that in release 6.6.0 Eventing Timers can be cancelled using the cancelTimer() function, or by creating a new Timer with the same reference as an existing Timer refer to xref:eventing-timers.adoc#limitations[Timers: Limitations]._
```

File: modules/eventing/pages/troubleshooting-best-practices.adoc [hunk 13]
```diff
@@ -9,6 +9,10 @@ When deploying (or resuming a paused function) a threshold is dynamically calcul
 
 [.out]`There are 104 eventing workers configured to run on 24 cores. A sizing exercise is recommended.`
 
+Typically you shouldn’t configure more than 4 x the number of physical cores (or 2 x # vCPUS) across all your Eventing functions.
```
Comment (simon-dew) [inline modules/eventing/pages/troubleshooting-best-practices.adoc]:
```suggestion
Typically you shouldn’t configure more than 4 × the number of physical cores (or 2 × the number of vCPUS) across all your Eventing functions.
```

Minor edits for readability.

File: modules/eventing/pages/eventing-curl-spec.adoc [hunk 14]
```diff
@@ -1,18 +1,25 @@
 = cURL
 :page-edition: Enterprise Edition
 
-The curl() function provides a way of interacting with external entities using HTTP.
+The curl() function provides a way of interacting with external entities via a REST endpoint using HTTP or HTTPS.
 
-Eventing functions can interact with external systems by using curl function to call their REST API. The possibility to interact with external systems opens a lot of new use cases, such as propagation of data changes to other systems, notifying the application about interesting events, enriching documents with data from external systems, and so on. This feature (introduced in 6.5) is both reliable and secure: the cURL calls are limited to a predefined set of URL bindings, for each binding we can specify authentication, encryption and certificate validation as necessary.  
+Eventing functions can interact with external systems by using curl function to call their REST API. The possibility to interact with external systems opens a lot of new use cases, such as
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-curl-spec.adoc]:
```suggestion
Eventing functions can interact with external systems by using the curl() function to call their REST API. The possibility to interact with external systems opens a lot of new use cases, such as:
```

Trying to distinguish between `curl()` (the Eventing function) and `cURL` (the underlying technology).

File: modules/eventing/pages/eventing-curl-spec.adoc [hunk 15]
```diff
@@ -1,18 +1,25 @@
...
+
+* Propagation of data changes to other systems.
+* Notifying the application about interesting events.
+* Enriching documents with data from external systems.
+* etc..
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-curl-spec.adoc]:
```suggestion
* etc... 
```

File: modules/eventing/pages/eventing-curl-spec.adoc [hunk 16]
```diff
@@ -1,18 +1,25 @@
...
 
-Eventing functions can interact with external systems by using curl function to call their REST API. The possibility to interact with external systems opens a lot of new use cases, such as propagation of data changes to other systems, notifying the application about interesting events, enriching documents with data from external systems, and so on. This feature (introduced in 6.5) is both reliable and secure: the cURL calls are limited to a predefined set of URL bindings, for each binding we can specify authentication, encryption and certificate validation as necessary.  
+Eventing functions can interact with external systems by using curl function to call their REST API. The possibility to interact with external systems opens a lot of new use cases, such as
+
+* Propagation of data changes to other systems.
+* Notifying the application about interesting events.
+* Enriching documents with data from external systems.
+* etc.. 
+
+This feature (introduced in 6.5) is both reliable and secure: the cURL calls are limited to a predefined set of URL bindings, for each binding we can specify authentication, encryption and certificate validation as necessary.
```
Comment (simon-dew) [inline modules/eventing/pages/eventing-curl-spec.adoc]:
```suggestion
This feature (introduced in 6.5) is both reliable and secure: the cURL calls are limited to a predefined set of URL bindings, and for each binding we can specify authentication, encryption, and certificate validation as necessary.  
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
LGTM overall, some minor suggestions.

**Comment by amarantha-k** [comment]
@simon-dew Thanks for the detailed review and suggestions on this large PR. The suggestions have been incorporated.

**Review by simon-dew — APPROVED** [review]

**Pull request merged**